crates/bevy_time/src/timer.rs

```diff
@@ -399,14 +403,36 @@ impl Timer {
     }
 }
 
+/// Specifies [`Timer`] behavior.
+#[derive(Debug, Clone, Copy, Reflect)]
+#[reflect(Default)]
+pub enum TimerMode {
+    /// Run once and stop.
+    Once,
+    /// Reset when finished.
+    Repeating,
+}
+
+impl TimerMode {
+    pub fn repeating(self) -> bool {
+        matches!(self, Self::Repeating)
+    }
```

Comment:
For simple enum it's simpler to use `Eq` intead of this kind of shortcut method.
See [this topic](https://users.rust-lang.org/t/whats-your-opinion-about-the-is-variant-pattern-on-enum/78421/2?u=tguichaoua).

Reply:
For this one, I was thinking about removing the `repeating` method and use `mode == TimerMode::Repeating` instead (at the call site).

Reply:
I feel that, for example in `set_mode`,
```
if !self.mode.repeating() && mode.repeating() && self.finished {
```
looks better than
```
if self.mode != TimerMode::Repeating && mode == TimerMode::Repeating && self.finished {
```
and I think that it makes sense to have utility methonds on `TimerMode` to get a boolean without having to do a comparison yourself.

Reply:
On the other hand, it would probably also make sense to use `assert_eq!` instead of `assert!` in tests.

Reply:
The problem with utility methods is that if you want to be consistant you have to add one for each variant for each enum. That is complex to maintain.
For fieldless enums using equality is more straightforward and seems to be the most used way in the rust community.